# Data Exploreation

## 

In [1]:
import pandas as pd

In [25]:
# train_label = pd.read_csv('data/train.csv', sep=',')
# train_label
train_label = pd.read_csv('data/train.csv', sep='\n')

train_label = train_label['ImageID,Labels,Caption'].str.split(',', expand=True)
train_label = train_label.loc[:, '0':'2']
train_label.columns = ["ImageID", "Labels", "Caption"]
train_label

# ids = df['ImageID']
# df[ids.isin(ids[ids.duplicated()])]


ImageID  Labels                                            Caption
0          0.jpg       1  "Woman in swim suit holding parasol on sunny d...
1          1.jpg    1 19  "A couple of men riding horses on top of a gre...
2          2.jpg       1  "They are brave for riding in the jungle on th...
3          3.jpg  8 3 13  "a black and silver clock tower at an intersec...
4          4.jpg   8 3 7  "A train coming to a stop on the tracks out si...
...          ...     ...                                                ...
29999  29995.jpg   8 1 2  "A picture of a truck that is in the middle of...
30000  29996.jpg       1  "A plate topped with a pizza being cut with a ...
30001  29997.jpg       1        "A man riding a snowboard on top of  snow."
30002  29998.jpg       1  "This photo shows people skiing in the mountai...
30003  29999.jpg       1  "Two young men playing soccer and fighting for...

[30004 rows x 3 columns]

Hre we have a potential issue in the ids, the total rows is 30004, while the image id is from 0 to 29999

In [27]:
test_label = pd.read_csv('data/test.csv', sep='\n')

test_label = test_label['ImageID,Caption'].str.split(',', expand=True)
test_label = test_label.loc[:, '0':'1']
test_label.columns = ["ImageID", "Caption"]
test_label

ImageID                                            Caption
0     30000.jpg  "A little girl waring a krispy kreme hat holdi...
1     30001.jpg  "A beautiful young woman holding an orange fri...
2     30002.jpg  "A group of people sitting on couch next to a ...
3     30003.jpg       "A person on a snowboard rides on the hill."
4     30004.jpg  "A man riding a skateboard with a helmet on in...
...         ...                                                ...
9995  39995.jpg  "A group of men riding surfboards riding a mas...
9996  39996.jpg  "A motorcycle parked next to a car in a parkin...
9997  39997.jpg          "a little boy that is playing with a wii"
9998  39998.jpg  "group of kids play Frisbee golf in the middle...
9999  39999.jpg  "A man in a gray jacket standing next to a wom...

[10000 rows x 2 columns]

## Check if each class is balanced (list the numer of sample's every class)

## calculate mean and std

In [ ]:
import torch